contracts/modules/common/BaseFeature.sol

```diff
@@ -0,0 +1,173 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.s
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../../wallet/IWallet.sol";
+import "../../infrastructure/IModuleRegistry.sol";
+import "../../infrastructure/storage/ILockStorage.sol";
+import "./IFeature.sol";
+import "../../../lib/other/ERC20.sol";
+import "./IVersionManager.sol";
+
+/**
+ * @title BaseFeature
+ * @notice Base Feature contract that contains methods common to all Feature contracts.
+ * @author Julien Niset - <julien@argent.xyz>, Olivier VDB - <olivier@argent.xyz>
+ */
+contract BaseFeature is IFeature {
+
+    // Empty calldata
+    bytes constant internal EMPTY_BYTES = "";
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    // The address of the Lock storage
+    ILockStorage internal lockStorage;
+    // The address of the Version Manager
+    IVersionManager internal versionManager;
+
+    event FeatureCreated(bytes32 name);
+    event FeatureInitialised(address wallet);
+
+    /**
+     * @notice Throws if the wallet is locked.
+     */
+    modifier onlyWhenUnlocked(address _wallet) {
+        require(!lockStorage.isLocked(_wallet), "BF: wallet locked");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the target wallet of the call.
+     */
+    modifier onlyWallet(address _wallet) {
+        require(msg.sender == _wallet, "BF: caller must be wallet");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the VersionManager.
+     */
+    modifier onlyVersionManager() {
+        require(msg.sender == address(versionManager), "BF: caller must be VersionManager");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the owner of the target wallet.
+     */
+    modifier onlyWalletOwner(address _wallet) {
+        require(isOwner(_wallet, msg.sender), "BF: must be wallet owner");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not an authorised feature of the target wallet.
+     */
+    modifier onlyWalletFeature(address _wallet) {
+        require(versionManager.isFeatureAuthorised(_wallet, msg.sender), "BF: must be a wallet feature");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the owner of the target wallet or the feature itself.
+     */
+    modifier onlyWalletOwnerOrFeature(address _wallet) {
+        // Wrapping in an internal method reduces deployment cost by avoiding duplication of inlined code
+        verifyOwnerOrAuthorisedFeature(_wallet, msg.sender);
```

Comment:
This is just an idea but the `Feature` part of this modifier is only there to let the `RelayerManager` call the other features. Other than that we don't want features to call each other and we know it is a security risk to have such a general check while we only want to address a particular use case.

Now that we store features in the singleton `VersionManager` we could add information about which feature can call the other features. This would make the wallet more secure at no extra gas cost, address TOB issue #23 without creating immutable dependencies between features.